### ISSUE TITLE
Manual restart required everytime network connectivity changes

### ISSUE THREAD

**Issue by jakirkham** [boot2docker/boot2docker#945]
With an existing container running before switching WiFi networks, switching to/from Ethernet, being disconnected, this problem appears every time.

**Comment by darend** [comment]
Also seeing the issue with 1.7.0. Happy to provide anything that would assist in diagnosing.

**Comment by charford** [comment]
I'm seeing this issue as well. It's similar to #824 and I believe is related (the resulting error is the same).

**Comment by jakirkham** [comment]
I posted some steps to get an ISO working from a PR made by @SvenDowideit on the other issue ( https://github.com/boot2docker/boot2docker/issues/824#issuecomment-116759440 ). I'd be interested to know if you guys see an improvement with this problem, as well, or not.

**Comment by jakirkham** [comment]
So, I don't believe this issue is solved yet. That being said, with the aforementioned fix, I don't have to destroy the VM every time I switch networks. Instead, I merely have to reboot the VM. Running containers continue to run after the VM reboot.

**Comment by pgkelley4** [comment]
I don't believe this is fixed. I am still seeing this on the latest boot2docker (1.7.1).

`boot2docker restart` fixes the issue, but is required every time your network connectivity changes!

**Comment by randerzander** [comment]
One workaround is to manually edit /etc/resolve.conf within each container, and within any new containers you start.

**Comment by jakirkham** [comment]
@randerzander, sounds like quite a hack. Also, the `/etc/resolve.conf` of the VM remains the same as the container's too (excepting the container's hostname).

That being said, if we were to change them, what would you recommend we change them too?

**Comment by berngp** [comment]
In my case this is related to the VirtualBox Network bridge that gets created between the host and the VM instance (the virtual instance running the docker daemon). One way to fix it is to identify the network bridge used by _boot to docker_ and restart it. For example, knowing that the bridge name is `vboxnet2` you can restart it by calling `sudo ifconfig vboxnet2 down` and bring it up with `sudo ifconfig vboxnet2 up`.

You can identify the bridge by running `VBoxManage list runningvms` and then `VBoxManage showvminfo <name of running vm>`, e.g:

```
$ VBoxManage list runningvms
"boot2docker-vm" {68c77456-872f-4f56-9caa-53d5eba3c1df}
"kitchen-hadoop-utils-default-centos-64_default_1438108356553_61963" {94cfd27f-31c0-41ce-8d80-3bce0b786612}
```

```
$ VBoxManage showvminfo boot2docker-vm | grep "Host-only Interface"
NIC 2:           MAC: 0800279AEFFE, Attachment: Host-only Interface 'vboxnet2', Cable connected: on, Trace: off (file: none), Type: virtio, Reported speed: 0 Mbps, Boot priority: 0, Promisc Policy: deny, Bandwidth group: none
```

Hope this works for others.

PS. This can happen to any virtual instance running inside Virtual Box and, at least in my case, can happen to other instances running through Virtual Box.

**Comment by jakirkham** [comment]
@berngp, thanks for sharing. Was really hoping this would help me, but it didn't.

**Comment by pencilcheck** [comment]
This is definitely a virtualbox issue as I still experience this issue in a different vm on virtualbox

**Comment by schmunk42** [comment]
On an Ubuntu installation `service network restart` could be a quick-fix, does someone know the equivalent for `boot2docker`?

**Comment by RomanSaveljev** [comment]
@schmunk42 If by `service network restart` you mean to re-new the lease, then killing `udhcpc` (there is 2 instances in b2d - one for every interface) with `SIGUSR1` should do it.

**Comment by jakirkham** [comment]
I think there was some PR that went into `docker-machine` that helps mitigate this with some settings, but I can't seem to find it now.

**Comment by schmunk42** [comment]
@RomanSaveljev Thank you! I'll check it, when I have the problem the next time.

The command below works, I don't know yet if it will fix the problem

    docker-machine ssh $(docker-machine active) "sudo udhcpc SIGUSR1 && sudo /etc/init.d/docker restart"

@jakirkham Is it a new command like `docker-machine update-dns`?

**Comment by jakirkham** [comment]
No, it was some options one could chose to give to the `create` subcommand. Found the PR ( https://github.com/docker/machine/pull/2467 ). Hopefully, that explains it a bit better.

**Comment by jakirkham** [comment]
I'll be moving around a lot today. So, I'll take this for a test run.

**Comment by schmunk42** [comment]
I just came home and had the described problem.

First run of my command above threw this error:

    Loading host from store failed: Host does not exist: "WARNING"

For whatever reason?!

But the second one worked and fixed the issue! :ok:

**Comment by jakirkham** [comment]
Yeah, so I tried creating a machine using `docker-machine` with the flags `--virtualbox-dns-proxy` and `--virtualbox-host-dns-resolver`. I was able to switch between several networks and never once lost internet connectivity nor did I have to reboot my VM for any reason. So, I think that is the solution to this issue.

**Comment by schmunk42** [comment]
Nice! 
These options are fairly new, I had 0.5.2. Now after an update, I can also see them :)

**Comment by jakirkham** [comment]
Yes, I think they both showed up in 0.5.3.

**Comment by jingles** [comment]
if you use virtualbox, you can pause the machine and resume it and it will attempt to restore outbound network connections.

<img width="251" alt="screen shot 2017-08-14 at 3 04 07 pm" src="https://user-images.githubusercontent.com/313998/29286965-e51854ba-8101-11e7-8dbf-09360db5cddf.png">
